lib/azure/armrest/insights/metrics_service.rb

```diff
@@ -0,0 +1,59 @@
+module Azure
+  module Armrest
+    module Insights
+      class MetricsService < ArmrestService
+        # Creates and returns a new MetricsService object.
+        #
+        def initialize(armrest_configuration, options = {})
+          options['api_version'] = '2014-04-01' # Must hard code for now
+          super(armrest_configuration, 'metricDefinitions', 'Microsoft.Insights', options)
+        end
+
+        # Return the metric definitions for the given +provider+, +resource_type+,
+        # and +resource_name+ for +resource_group+. You may pass a :filter option as well.
+        #
+        # Example:
+        #
+        #   metrics = Azure::Armrest::Insights::MetricsService.new(conf)
+        #
+        #   metrics.list('Microsoft.SQL', 'servers', 'myServer/databases/myDB', 'mygroup')
+        #   metrics.list('Microsoft.Compute', 'virtualMachines', 'myVm', 'mygroup')
+        #
+        def list(provider, resource_type, resource_name, resource_group = nil, options = {})
+          resource_group ||= configuration.resource_group
+
+          raise ArgumentError, "no resource group provided" unless resource_group
+
+          url = build_url(provider, resource_type, resource_name, resource_group, options)
+
+          response = rest_get(URI.escape(url))
```

Comment:
why do we need `URI.escape` here? should the same be applied for other `rest_get`?